fatrace.c

```diff
@@ -307,9 +307,11 @@ print_event (const struct fanotify_event_metadata *data,
         if (len < 0) {
             /* fall back to the device/inode */
             struct stat st;
-            if (fstat (event_fd, &st) < 0)
-                err (EXIT_FAILURE, "stat");
-            snprintf (pathname, sizeof (pathname), "device %i:%i inode %ld\n", major (st.st_dev), minor (st.st_dev), st.st_ino);
+            if (fstat(event_fd, &st) < 0) {
```

Comment:
Copying from [previous review](https://github.com/martinpitt/fatrace/pull/49#discussion_r2176444585).

Did you see this happen in real life? strace runs as root (so shouldn't get EPERM), and neither ENOENT as the kernel does not actually remove files while they still have an open fd. I don't want to hide logic errors that way, so can you please at least put a `warn("stat")` here?

Reply:
I believe I've seen this in the wild, it had something to do with short-lived processes, but I can't reproduce it easily. I've added the warning.

Reply:
The problem came up during developing `--json`, where I do this unconditionally on all events. Since it happened there, I surmised it could maybe also happen conditionally, when reading path has already failed.